src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeAwaitable.cs

```diff
@@ -50,13 +50,17 @@ public void BeginOperation(CancellationToken cancellationToken, Action<object?>
             // Don't register if already completed, we would immediately unregistered in ObserveCancellation
             if (cancellationToken.CanBeCanceled && !IsCompleted)
             {
+                var previousAwaitableState = _awaitableState;
```

Comment:
Should this be a `#if DEBUG` so we don't have this field copy in release builds?